ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -614,6 +614,28 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
         const refreshing = app.metadata.annotations && app.metadata.annotations[appModels.AnnotationRefreshKey];
         const fullName = AppUtils.nodeKey({group: 'argoproj.io', kind: app.kind, name: app.metadata.name, namespace: app.metadata.namespace});
         const ActionMenuItem = (prop: {actionLabel: string}) => <span className={needOverlapLabelOnNarrowScreen ? 'show-for-large' : ''}>{prop.actionLabel}</span>;
+        const resourceActions = services.applications
```

Comment:
I basically copied this block from the function that builds the resource node menus, just changing the server-hitting requests for application action getters/runners.